lesson11/index.js

```diff
@@ -0,0 +1,77 @@
+const ul = document.getElementById('js-list');
+
+const createLoading = () => {
+    const li = document.createElement("li");
+    const loadingImg = document.createElement('img');
+
+    li.style.listStyle = "none";
+    li.id = "js-loading";
+    loadingImg.src = 'loading-circle.gif';
+    ul.appendChild(li).appendChild(loadingImg);
+};
+
+const removeLoading = () => document.getElementById('js-loading').remove();
+
+const addList = (val) => {
+    const fragment = document.createDocumentFragment();
+    for (let i = 0; i < val.length; i++) {
+        const li = document.createElement('li');
+        const a = document.createElement('a');
+        const img = document.createElement('img');
+    
+        a.href = `${val[i].a}.html`;
+        a.textContent = `${val[i].text}`;
+        img.src = val[i].img;
+        img.alt = val[i].alt;
+    
+        fragment.appendChild(li).appendChild(a).appendChild(img);
+    }
+
+    ul.appendChild(fragment);
+};
+
+const createErrorMessage = (error) => {
+    const li = document.createElement("li");
+    li.textContent =`エラー内容：${error.message}`;
+    ul.appendChild(li);
+    console.error(error.message);
+};
+
+/**
+ * Async関数 
+ * @return {Array} 
+ */
+const jsonUrl = 'https://myjson.dit.upm.es/api/bins/4ubl';
+const fetchData = async () => {
+    try {
+        const response = await fetch(jsonUrl) // 中身はresponseオブジェクトを含むpromise
+        const json = await response.json(); //response オブジェクトから JSON を抽出
+        return json.data;
+
+    } catch(error) {
+        console.error(error);
+    } 
+};
+
+/**
+ * Async関数
+ * fetchData関数から値が返ってきた後、addList関数にその値を渡す
+ */
+const createDataList = async () => {
+    createLoading();
+    try {
+        const data = await fetchData();
+        if( data.length === 0 ) {
+            throw new Error('データが空です。');
+        }
+        addList(data);
+
+    } catch(error) {
+        createErrorMessage(error);
+
+    }  finally {
+        removeLoading();
+    }
+};
```

Comment:
Thanks for fixing that in such a short time!
I think it turned out very well!

but I would like to comment on one point I noticed in createDataList.

I felt uncomfortable with the flow of executing addList() in try and then finally removing the loading image in finally.

After all, shouldn't try, catch, and finally here be functions that get the data of the list?
I think it would be better to execute addList() after receiving the data.

Reply:
We have reflected the feedback we received.
[f0b94c9](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/f0b94c91c355db22d68f56e590feabd7d5aac024)
[a20503b](https://github.com/AyakaOnozuka131/JavaScript-lessons/pull/11/commits/a20503b2922fd9e6a99207e7ebc6773d5fb04b72)